### ISSUE TITLE
PostgresCluster switchover on pgo upgrade (5.1.2 -> 5.2.0)

### ISSUE THREAD

**Issue by pere3** [CrunchyData/postgres-operator#3428]
Hello, crunchy community and crunchy development team,

recently i've updated our pgo installation from 5.1.2 to 5.20 with a few PostgresCluster's already in production under pgo.

During the process of this update i've noticed that all of PostgresCluster's executed switchovers while new pgo operator started.

I wonder what is causing PostgresCluster's to perform switchovers on pgo upgrade (even minor 5.1.2 -> 5.2.0)?
I understand that this pgo version introduced CRD update, but there will be CRD updates in the future and by the time being we can have hundreds of PostgresCluster's that will perform switchovers simultaneously with a service interruption in applications using these databases.

This almost sounds like a showstopper, with potential downtime on a lot of clusters simultaneously with every future operator upgrade.

Is this behavior controllable?

**Comment by pere3** [comment]
@andrewlecuyer i'm sorry for direct mention, but can you please comment on this? it's crucial for our adoption process

**Comment by andrewlecuyer** [comment]
@pere3 I recommend checking out the new "pause" feature that is included in PGO v5.2.0:

https://access.crunchydata.com/documentation/postgres-operator/latest/tutorial/administrative-tasks/#pausing-reconciliation-and-rollout

This allows you to pause reconciliation for a PostgresCluster using the new `spec.paused` field, which can be leveraged to control rollouts during an upgrade.  We actually just published a blog that describes similar use of pause during an upgrade:

https://www.crunchydata.com/blog/easier-upgrades-and-image-management-for-postgres-in-kubernetes

Also, please note that because the PostgresCluster CRD is backwards compatible (as are all PGO CRDs), you can first update the CRD and set `spec.paused` on your various PostgresCluser's _prior_ to upgrading the `pgo` Deployment (which will:

```bash
$ kubectl apply --server-side -k kustomize/install/crd
customresourcedefinition.apiextensions.k8s.io/pgupgrades.postgres-operator.crunchydata.com serverside-applied
customresourcedefinition.apiextensions.k8s.io/postgresclusters.postgres-operator.crunchydata.com serverside-applied
$ kubectl patch postgrescluster/hippo -n postgres-operator --type merge   --patch '{"spec":{"paused": true}}'
postgrescluster.postgres-operator.crunchydata.com/hippo patched
$ kubectl apply --server-side -k kustomize/install/default
customresourcedefinition.apiextensions.k8s.io/pgupgrades.postgres-operator.crunchydata.com serverside-applied
customresourcedefinition.apiextensions.k8s.io/postgresclusters.postgres-operator.crunchydata.com serverside-applied
serviceaccount/pgo serverside-applied
serviceaccount/postgres-operator-upgrade serverside-applied
clusterrole.rbac.authorization.k8s.io/postgres-operator serverside-applied
clusterrole.rbac.authorization.k8s.io/postgres-operator-upgrade serverside-applied
clusterrolebinding.rbac.authorization.k8s.io/postgres-operator serverside-applied
clusterrolebinding.rbac.authorization.k8s.io/postgres-operator-upgrade serverside-applied
deployment.apps/pgo serverside-applied
deployment.apps/pgo-upgrade serverside-applied
```

**Comment by pere3** [comment]
Thank you so much for this answer and a blog post, that is actually very useful information!
